src/containers/profile/ContractSettings.js

```diff
@@ -3,20 +3,28 @@ import classNames from 'classnames';
 import { Row, Col } from 'reactstrap';
 import { Desktop, Mobile } from '../../generic/MediaQuery';
 import {FormattedMessage, injectIntl} from 'react-intl';
-import {showInfoModal} from '../../actions/modal';
+import {disableTwoFactorAuthModal, enableTwoFactorAuthModal, showInfoModal} from '../../actions/modal';
 import {connect} from 'react-redux';
+import {getProfilePageInfo} from '../../actions/profile';
 
 class ContractSettings extends React.Component {
 
   constructor(props) {
     super(props);
     this.onToggleClick = this.onToggleClick.bind(this);
-    this.state = {isEditing: false };
+    this.state = {isEditing: false,  is2FAEnable: this.props.profile.mfaEnabled || false} ;
```

Comment:
2fa should not be implemented in ContractSettings component. There should be separated component for that like SecuritySettings.